src/Workspaces/Core/Portable/Storage/SQLite/SQLitePersistentStorage.Accessor.cs

```diff
@@ -68,86 +74,74 @@ public async Task<Checksum> ReadChecksumAsync(TKey key, CancellationToken cancel
                 return null;
             }
 
+            [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36114", AllowCaptures = false)]
             public Task<Stream> ReadStreamAsync(TKey key, Checksum checksum, CancellationToken cancellationToken)
-                => ReadBlobColumnAsync(key, DataColumnName, checksum, cancellationToken);
+                => Storage.PerformReadAsync(s_readStream, (self: this, key, checksum, cancellationToken), cancellationToken);
+
+            private static readonly Func<(Accessor<TKey, TWriteQueueKey, TDatabaseId> self, TKey key, Checksum checksum, CancellationToken cancellationToken), Stream> s_readStream =
+                t => t.self.ReadStream(t.key, t.checksum, t.cancellationToken);
 
-            private async Task<Stream> ReadBlobColumnAsync(
+            [PerformanceSensitive("https://github.com/dotnet/roslyn/issues/36114", AllowCaptures = false)]
+            private Stream ReadStream(TKey key, Checksum checksum, CancellationToken cancellationToken)
+                => ReadBlobColumn(key, DataColumnName, checksum, cancellationToken);
+
+            private Stream ReadBlobColumn(
                 TKey key, string columnName, Checksum checksumOpt, CancellationToken cancellationToken)
             {
                 cancellationToken.ThrowIfCancellationRequested();
 
                 if (!Storage._shutdownTokenSource.IsCancellationRequested)
```

Comment:
i've kept this check around.  But it should actually be an invariant failure.  It would be 100% busted for the storage service to be shutdown and then have some consumer try to read from it.